src/main/java/org/openstreetmap/atlas/geography/atlas/change/FeatureChangeMergingHelpers.java

```diff
@@ -272,6 +109,92 @@ else if (afterEntityLeft instanceof Relation)
+            final Node beforeNodeRight = (Node) beforeEntityRight;
+            final CompleteNode mergedBeforeNode = CompleteNode.from(beforeNodeLeft);
+            final SortedSet<Long> leftInEdgeIdentifiers = new TreeSet<>(beforeNodeLeft.inEdges()
+                    .stream().map(Edge::getIdentifier).collect(Collectors.toSet()));
+            final SortedSet<Long> rightInEdgeIdentifiers = new TreeSet<>(beforeNodeRight.inEdges()
+                    .stream().map(Edge::getIdentifier).collect(Collectors.toSet()));
+            final SortedSet<Long> leftOutEdgeIdentifiers = new TreeSet<>(beforeNodeLeft.outEdges()
+                    .stream().map(Edge::getIdentifier).collect(Collectors.toSet()));
+            final SortedSet<Long> rightOutEdgeIdentifiers = new TreeSet<>(beforeNodeRight.outEdges()
+                    .stream().map(Edge::getIdentifier).collect(Collectors.toSet()));
+
+            if (!leftInEdgeIdentifiers.equals(rightInEdgeIdentifiers))
+            {
+                mergedBeforeNode.withInEdgeIdentifiers(
+                        MemberMergeStrategies.simpleLongSortedSetAllowCollisionsMerger
+                                .apply(leftInEdgeIdentifiers, rightInEdgeIdentifiers));
+            }
+            if (!leftOutEdgeIdentifiers.equals(rightOutEdgeIdentifiers))
+            {
+                mergedBeforeNode.withOutEdgeIdentifiers(
+                        MemberMergeStrategies.simpleLongSortedSetAllowCollisionsMerger
+                                .apply(leftOutEdgeIdentifiers, rightOutEdgeIdentifiers));
+            }
+            return new FeatureChange(ChangeType.REMOVE, left.getAfterView(), mergedBeforeNode);
+        }
+        /*
+         * For relations, we need to merge the beforeViews of the RelationBean and
+         * allKnownOsmMembersBean.
+         */
+        else if (beforeEntityLeft instanceof Relation)
+        {
+            final Relation beforeRelationLeft = (Relation) beforeEntityLeft;
+            final Relation beforeRelationRight = (Relation) beforeEntityRight;
+            final CompleteRelation mergedBeforeRelation = CompleteRelation.from(beforeRelationLeft);
+            final RelationBean leftMembers = beforeRelationLeft.members().asBean();
+            final RelationBean rightMembers = beforeRelationRight.members().asBean();
+            final RelationBean leftOsmMembers = beforeRelationLeft.allKnownOsmMembers().asBean();
+            final RelationBean rightOsmMembers = beforeRelationRight.allKnownOsmMembers().asBean();
+
+            if (!leftMembers.equalsIncludingExplicitlyExcluded(rightMembers))
+            {
+                mergedBeforeRelation.withMembers(RelationBean.mergeBeans(leftMembers, rightMembers),
+                        Rectangle.forLocated(beforeRelationLeft, beforeRelationRight));
+            }
+            if (!leftOsmMembers.equalsIncludingExplicitlyExcluded(rightOsmMembers))
+            {
+                mergedBeforeRelation.withAllKnownOsmMembers(
```

Comment:
Do we actually need to merge `allKnownOsmMembers`? I'm not sure if this field is ever inconsistent across shards. It sounds like it isn't.

Reply:
It looks like it is not.